### PR TITLE
Feature registration

### PR DESCRIPTION
**Pull request by mijicuros00**

### CODE REVIEW

File: social-network/src/main/java/com/levi9/socialnetwork/Config/WebSecurityConfig.java [hunk 1]
```diff
@@ -67,7 +70,7 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
 				// .antMatchers("/admin").hasAuthority("ROLE_ADMIN")
```
Comment (daleksic) [inline social-network/src/main/java/com/levi9/socialnetwork/Config/WebSecurityConfig.java]: If this line is not needed, remove it.

File: social-network/src/main/java/com/levi9/socialnetwork/Controller/AuthController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Autowired
+    private RegistrationService registrationService;
+
+    @PostMapping(value = "/registration")
+    public ResponseEntity<Void> registration(@RequestBody RegistrationRequestDTO registrationRequestDTO){
+
+        try{
+            registrationService.register(registrationRequestDTO);
+        }catch (IllegalStateException illegalStateException){
```
Comment (daleksic) [inline social-network/src/main/java/com/levi9/socialnetwork/Controller/AuthController.java]: Missing space after }, and before {

File: social-network/src/main/java/com/levi9/socialnetwork/Controller/AuthController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+@RequestMapping("api/auth")
+public class AuthController {
+
+    @Autowired
+    private RegistrationService registrationService;
+
+    @PostMapping(value = "/registration")
+    public ResponseEntity<Void> registration(@RequestBody RegistrationRequestDTO registrationRequestDTO){
+
+        try{
```
Comment (daleksic) [inline social-network/src/main/java/com/levi9/socialnetwork/Controller/AuthController.java]: Missing space before {

File: social-network/src/main/java/com/levi9/socialnetwork/Model/ConfirmationToken.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package com.levi9.socialnetwork.Model;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
```
Comment (daleksic) [inline social-network/src/main/java/com/levi9/socialnetwork/Model/ConfirmationToken.java]: Import only classes that you need from libraries.

File: social-network/src/main/java/com/levi9/socialnetwork/Model/EStatus.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.levi9.socialnetwork.Model;
+
+public enum EStatus {
```
Comment (daleksic) [inline social-network/src/main/java/com/levi9/socialnetwork/Model/EStatus.java]: EStatus name is unclear what is this enum about. Try to rename it.

File: social-network/src/main/java/com/levi9/socialnetwork/Service/impl/EmailServiceImpl.java [hunk 6]
```diff
@@ -30,15 +37,99 @@ public void sendNotificaitionAsync(User user) throws MailException, InterruptedE
...
+			javaMailSender.send(mimeMessage);
+		} catch (MessagingException e) {
+			e.printStackTrace();
+			throw new IllegalStateException("Failed to send email");
+		}
+	}
+
+	@Override
+	public String registerEmail(String name, String link) {
+		return "<div style=\"font-family:Helvetica,Arial,sans-serif;font-size:16px;margin:0;color:#191919\">\n" +
```
Comment (daleksic) [inline social-network/src/main/java/com/levi9/socialnetwork/Service/impl/EmailServiceImpl.java]: You can extract this template into a separate html resource file and populate template fields here. If you have time.

File: social-network/src/main/java/com/levi9/socialnetwork/Service/impl/RegistrationServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+            Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);
+
+    public static boolean validate(String emailStr) {
+        Matcher matcher = VALID_EMAIL_ADDRESS_REGEX.matcher(emailStr);
+        return matcher.find();
+    }
+
+    public static boolean validateRegistrationRequest(RegistrationRequestDTO registrationRequestDTO){
+        if (registrationRequestDTO.getName().isBlank() || registrationRequestDTO.getSurname().isBlank() ||
+                registrationRequestDTO.getPassword().length() < 6 || registrationRequestDTO.getRepeatedPassword().length() < 6 ||
```
Comment (daleksic) [inline social-network/src/main/java/com/levi9/socialnetwork/Service/impl/RegistrationServiceImpl.java]: Extract 6 as constant PASSWORD_LENGTH or something similar. If a change of length is needed, you will change it only in one place.

File: social-network/src/main/java/com/levi9/socialnetwork/Service/impl/RegistrationServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+    @Transactional
+    @Override
+    public String register(RegistrationRequestDTO registrationRequestDTO) {
+        boolean isRequestValid = validateRegistrationRequest(registrationRequestDTO);
+        if(!isRequestValid){
+            throw new IllegalStateException("Registration request is not valid!");
+        }
+
+        User user;
+        try{
```
Comment (daleksic) [inline social-network/src/main/java/com/levi9/socialnetwork/Service/impl/RegistrationServiceImpl.java]: Add missing spaces after or before {} in the whole try...catch block

File: social-network/src/main/java/com/levi9/socialnetwork/Service/impl/UserServiceImpl.java [hunk 9]
```diff
@@ -59,7 +59,17 @@ public ResponseEntity<User> getUserById(Long userId)
 		User user = userRepository.findById(userId).orElseThrow(() -> new ResourceNotFoundException("User not found for this id :: " + userId));
 	 return ResponseEntity.ok().body(user);
 	}
-	
+
+	@Override
+	public User findUserById(Long userId) throws ResourceNotFoundException {
+		return userRepository.findById(userId).orElseThrow(() -> new ResourceNotFoundException("User not found for this id :: " + userId));
```
Comment (daleksic) [inline social-network/src/main/java/com/levi9/socialnetwork/Service/impl/UserServiceImpl.java]: Extract the message in a constant and use it here and in getUserById(Long userId).

File: social-network/src/main/java/com/levi9/socialnetwork/Model/EUserVerifiedRegistration.java [hunk 10]
```diff
@@ -0,0 +1,6 @@
+package com.levi9.socialnetwork.Model;
+
+public enum EUserVerifiedRegistration {
```
Comment (daleksic) [inline social-network/src/main/java/com/levi9/socialnetwork/Model/EUserVerifiedRegistration.java]: Naming is the biggest problem of programmers :) 
I would name it something like UserVerificationStatus, no need for E

File: social-network/src/main/java/com/levi9/socialnetwork/Service/impl/RegistrationServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+            Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);
+
+    public static boolean validate(String emailStr) {
+        Matcher matcher = VALID_EMAIL_ADDRESS_REGEX.matcher(emailStr);
+        return matcher.find();
+    }
+
+    public static boolean validateRegistrationRequest(RegistrationRequestDTO registrationRequestDTO){
+        if (registrationRequestDTO.getName().isBlank() || registrationRequestDTO.getSurname().isBlank() ||
+                registrationRequestDTO.getPassword().length() < PASSWORD_MIN_LENGTH || registrationRequestDTO.getRepeatedPassword().length() < 6 ||
```
Comment (daleksic) [inline social-network/src/main/java/com/levi9/socialnetwork/Service/impl/RegistrationServiceImpl.java]: Use it also for the second 6

File: social-network/src/main/java/com/levi9/socialnetwork/Model/EUserVerifiedRegistration.java (hunk 10 above)
Comment (mijicuros00) [reply social-network/src/main/java/com/levi9/socialnetwork/Model/EUserVerifiedRegistration.java]: I agree, that is much better and more understandable. I updated the name.

File: social-network/src/main/java/com/levi9/socialnetwork/Service/impl/EmailServiceImpl.java (hunk 6 above)
Comment (mijicuros00) [reply social-network/src/main/java/com/levi9/socialnetwork/Service/impl/EmailServiceImpl.java]: Extracted HTML code to separate file and then populated it with data. Used apache commons for working with html file.

**Pull request merged**